mopidy_webhooks/reporters/status.py

```diff
@@ -22,39 +17,46 @@ class StatusReporter(pykka.ThreadingActor):
     """
 
     def __init__(self, config, core):
-        super(StatusReporter, self).__init__()
-        self.config = config['webhooks']
+        super().__init__()
+        self.config = config["webhooks"]
         self.core = core
         self.in_future = self.actor_ref.proxy()
 
     def on_start(self):
-        """Runs when the actor is started and schedules a status update
-        """
-        logger.info('StatusReporter started.')
-        # if configured not to report status then return immediately
-        if self.config['status_update_interval'] == 0:
-            logger.info('StatusReporter disabled by configuration.')
-            return
+        """Runs when the actor is started and schedules a status update"""
+        logger.info("Webhooks StatusReporter started.")
+
         self.in_future.report_status()
 
-    def report_again(self, current_status):
+    def report_again(self, playback_state):
         """Computes a sleep interval, sleeps for the specified amount of time
         then kicks off another status report.
         """
-        # calculate sleep interval based on current status and configured interval
-        _m = {'playing': 1, 'paused': 2, 'stopped': 5}[current_status['state']]
-        interval = (self.config['status_update_interval'] * _m) / 1000.0
-        # sleep for computed interval and kickoff another webhook
+        # Calculate sleep interval based on current playback state and configured interval.
+        _m = {"playing": 1, "paused": 2, "stopped": 5}[playback_state]
+        interval = (self.config["status_update_interval"] * _m) / 1000.0
+
+        # Sleep for computed interval, then kickoff another webhook.
         time.sleep(interval)
         self.in_future.report_status()
 
     def report_status(self):
-        """Get status of player from mopidy core and send webhook.
-        """
+        """Get status of player from Mopidy core and send webhook."""
+        playback_state_future = self.core.playback.get_state()
+        current_track_future = self.core.playback.get_current_track()
+        time_position = self.core.playback.get_time_position()
+
+        playback_state = playback_state_future.get()
+
         current_status = {
-            'current_track': self.core.playback.current_track.get(),
-            'state': self.core.playback.state.get(),
-            'time_position': self.core.playback.time_position.get(),
+            "state": playback_state,
+            "current_track": current_track_future.get(),
+            "time_position": time_position.get(),
         }
-        send_webhook(self.config, {'status_report': current_status})
-        self.report_again(current_status)
+        payload = {
+            "type": "status",
+            "data": current_status,
+        }
+
+        send_webhook(self.config, payload)
+        self.report_again(playback_state)
```

Comment:
Hopefully when this PR is merged, this code can either be replaced with a single call to `core.get_status()`, or a second function named something like "full status" could be added to do that instead.

https://github.com/mopidy/mopidy/pull/2007